packages/vite-plugin-svelte/src/utils/watch.ts

```diff
@@ -33,9 +41,38 @@ export function setupWatchers(
 		}
 	};
 
-	watcher.on('change', emitChangeEventOnDependants);
-	watcher.on('unlink', (filename) => {
-		removeUnlinkedFromCache(filename);
-		emitChangeEventOnDependants(filename);
+	const possibleSvelteConfigs = knownSvelteConfigNames.map((cfg) => path.join(root, cfg));
+	const restartOnConfigAdd = (filename: string) => {
+		if (possibleSvelteConfigs.includes(filename)) {
+			log.info(`svelte config added: restarting vite server. - file: ${filename}`);
+			server.watcher.emit('change', viteConfig);
+		}
+	};
+
+	const restartOnConfigChange = (filename: string) => {
+		if (filename === svelteConfigFile) {
```

Comment:
bug: `svelteConfigFile` here needs to be an absolute path to possibly match this condition, otherwise custom config files doesn't work.

Reply:
https://github.com/sveltejs/vite-plugin-svelte/blob/4e876d2c61f6f8b009618aa84c7b3bdd4bc30d55/packages/vite-plugin-svelte/src/utils/load-svelte-config.ts#L73

should be absolute, the path is resolved and updated while loading

Reply:
It isn't absolute for `packages/vite-plugin-svelte/src/utils/watch.ts` though, through some logging the value was always e.g. `svelte.config.custom.cjs`. I'm currently testing it at the `configfile-custom` e2e test package, and the config change doesn't trigger a server restart.